### PR TITLE
Adapt Java bindings to latest changes in Explicit Disclosure

### PR DESCRIPTION
**Pull request by tudor-da**
This PR adapts the Java bindings to the recent changes in the Explicit Disclosure Ledger API interfaces ([source](https://github.com/digital-asset/daml/commit/25ae8d2281a8f34473dd9e3ba9d11706b83eb64f))

### CODE REVIEW

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java [hunk 1]
```diff
@@ -44,16 +53,74 @@ public TransactionFilterOuterClass.InterfaceFilter toProto(Identifier interfaceI
...
+          && Objects.equals(includeCreateEventPayload, that.includeCreateEventPayload);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(includeInterfaceView, includeCreateEventPayload);
+    }
+  }
+
+  public static final class Template {
```
Comment (raphael-speyer-da) [inline language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:91]: Also probably better as an enum.

```suggestion
  public static enum Template {
    INCLUDE_PAYLOAD(true),
    HIDE_PAYLOAD(false);
```

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java [hunk 2]
```diff
@@ -18,22 +19,30 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
...
+   * #HIDE_VIEW_INCLUDE_PAYLOAD} and {@link #INCLUDE_VIEW_INCLUDE_PAYLOAD}.
    */
   public static final class Interface {
     public final boolean includeInterfaceView;
-    // add equals and hashCode if adding more fields
+    public final boolean includeCreateEventPayload;
 
-    public static final Interface INCLUDE_VIEW = new Interface(true);
-    public static final Interface HIDE_VIEW = new Interface(false);
+    public static final Interface HIDE_VIEW_HIDE_PAYLOAD = new Interface(false, false);
```
[the remark this answers is not in the document]
Comment (mziolekda) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:30]: the rename to `created_event_blob` is about to hit the repo, so then these fields should follow suit.

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DisclosedContract.java [hunk 3]
```diff
@@ -5,27 +5,46 @@
...
+import com.google.protobuf.ByteString;
+import java.util.Optional;
 
 public final class DisclosedContract {
   public final Identifier templateId;
   public final String contractId;
-  public final Arguments arguments;
-  public final ContractMetadata contractMetadata;
+  public final Optional<Arguments> arguments;
+  public final Optional<ContractMetadata> contractMetadata;
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/DisclosedContract.java]: Yes, proceeded as suggested, thanks!

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java]: I've went the verbose way in the end for expressiveness (`*_CREATED_EVENT_BLOB`)

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java (hunk 1 above)
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java]: Done

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java [hunk 4]
```diff
@@ -18,22 +19,30 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
   public abstract TransactionFilterOuterClass.Filters toProto();
 
   /**
-   * Settings for including an interface in {@link InclusiveFilter}. There are two values: {@link
-   * #INCLUDE_VIEW} and {@link #HIDE_VIEW}.
+   * Settings for including an interface in {@link InclusiveFilter}. There are four possible values:
+   * {@link #HIDE_VIEW_HIDE_PAYLOAD} and {@link #INCLUDE_VIEW_HIDE_PAYLOAD} and {@link
+   * #HIDE_VIEW_INCLUDE_PAYLOAD} and {@link #INCLUDE_VIEW_INCLUDE_PAYLOAD}.
    */
   public static final class Interface {
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java]: Done

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java [hunk 5]
```diff
@@ -70,6 +75,72 @@ public CreatedEvent(
...
+        signatories,
+        observers);
+  }
+
+  /**
+   * @deprecated Passing {@code createArgumentsBlob} and {@code contractMetadata} arguments is
+   *     deprecated. Use the constructor with {@link #createEventPayload} instead. Since Daml 2.8.0
+   */
+  @Deprecated
+  public CreatedEvent(
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CreatedEvent.java]: Indeed, went without backwards compatibility support here (see ad242ec8d8c679ac62f12eb10a0759b5790b7d3a)

**Comment by tudor-da** [comment]
@raphael-speyer-da Thanks for the review. I've addressed the review comments but generally I've changed some DTO's to non-backwards compatibility (`CreatedEvent` and `DisclosedContract`) since the affected fields haven't indeed been released in a GA feature. Please re-review

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java [hunk 6]
```diff
@@ -18,22 +18,33 @@ public static Filter fromProto(TransactionFilterOuterClass.Filters filters) {
...
-   * #INCLUDE_VIEW} and {@link #HIDE_VIEW}.
+   * Settings for including an interface in {@link InclusiveFilter}. There are four possible values:
+   * {@link #HIDE_VIEW_HIDE_CREATED_EVENT_BLOB} and {@link #INCLUDE_VIEW_HIDE_CREATED_EVENT_BLOB}
+   * and {@link #HIDE_VIEW_INCLUDE_CREATED_EVENT_BLOB} and {@link
+   * #INCLUDE_VIEW_INCLUDE_CREATED_EVENT_BLOB}.
    */
-  public static final class Interface {
-    public final boolean includeInterfaceView;
-    // add equals and hashCode if adding more fields
+  public enum Interface {
```
Comment (raphael-speyer-da) [inline language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:26]: These types should probably be `static`.

```suggestion
  public static enum Interface {
```

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java [hunk 7]
```diff
@@ -44,16 +55,43 @@ public TransactionFilterOuterClass.InterfaceFilter toProto(Identifier interfaceI
...
 
     Interface merge(Interface other) {
-      return includeInterfaceView(includeInterfaceView || other.includeInterfaceView);
+      return includeInterfaceView(
+          includeInterfaceView || other.includeInterfaceView,
+          includeCreatedEventBlob || other.includeCreatedEventBlob);
+    }
+  }
+
+  public enum Template {
```
Comment (raphael-speyer-da) [inline language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:69]:
```suggestion
  public static enum Template {
```

**Review by raphael-speyer-da — APPROVED** [review]
LGTM modulo comments

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/InclusiveFilter.java [hunk 8]
```diff
@@ -28,15 +45,23 @@ public InclusiveFilter(@NonNull Set<@NonNull Identifier> templateIds) {
     this(templateIds, Collections.emptyMap());
   }
 
+  /**
+   * @deprecated Use the constructor without with {@link #templateFilters} instead IDs. Since Daml
```
[the remark this answers is not in the document]
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/InclusiveFilter.java:49]: Apologies, applied but didn't pull after 🤦

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java (hunk 6 above)
Comment (tudor-da) [reply language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/Filter.java:26]: IJ was suggesting that is not necessary for inner enums. But fine by me to apply

**Pull request merged**